code/datums/datum.dm

```diff
@@ -449,18 +506,21 @@
 	. = FALSE
 	for(var/name in names)
 		if(filter_data[name])
+			qdel(filter_data[name])
 			filter_data -= name
-			. = TRUE
-
-	if(.)
-		update_filters()
-	return .
+	update_filters()
+	UNSETEMPTY(filter_data)
 
 /datum/proc/clear_filters()
 	ASSERT(isatom(src) || isimage(src))
-	var/atom/atom_cast = src // filters only work with images or atoms.
-	filter_data = null
-	atom_cast.filters = null
```

Comment:
Should probably keep `atom_cast.filters = null` here unless it's doing harm. I think a few places still mutate the filters list directly.

Reply:
We have the api for a reason it's their problem
Realized update filters should be called here anyway since it wasnt clearing properly and did that (it has = null inside)

Reply:
if something mutates the list directly you'd want it to not be mutated here either, so I'm with tivi on this one, this is correct.